src/polymer/polymer-core-feature.ts

```diff
@@ -12,9 +12,88 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {ScannedPolymerElement} from '../polymer/polymer-element';
+import {Class} from '../model/class';
+import {Document, Feature, Method, Property, Resolvable, ScannedFeature, ScannedMethod, ScannedProperty, Warning} from '../model/model';
 
 /**
- * The metadata for a Polymer core feature.
+ * A scanned Polymer 1.x core "feature".
  */
-export interface ScannedPolymerCoreFeature extends ScannedPolymerElement {}
+export class ScannedPolymerCoreFeature extends ScannedFeature implements
+    Resolvable {
+  warnings: Warning[] = [];
+  properties: ScannedProperty[] = [];
+  methods: ScannedMethod[] = [];
+
+  resolve(document: Document): Feature|undefined {
+    if (this.warnings.length > 0) {
+      return;
+    }
+
+    // Sniff for the root `_addFeatures` call by presence of this method. This
+    // will normally be in the `polymer.html` document.
+    const isRootAddFeatureCall =
+        this.methods.some((p) => p.name === '_finishRegisterFeatures');
+
+    if (!isRootAddFeatureCall) {
+      // We're at the `Polymer.Base` assignment or a regular `_addFeature`
+      // call. We'll merge all of these below.
+      return new PolymerCoreFeature(this.properties, this.methods);
+
+    } else {
+      // We're at the root of the `_addFeatures` tree. Thanks to HTML imports,
+      // we can be sure that this is the final core feature we'll resolve. Now
+      // we'll emit a "fake" class representing the union of all the feature
+      // objects.
+      const allProperties: Property[] = this.properties.slice();
+      const allMethods: Method[] = this.methods.slice();
+      const otherCoreFeatures = document.getFeatures({
+        kind: 'polymer-core-feature',
+        imported: true,
+        externalPackages: true,
+      });
+      for (const feature of otherCoreFeatures) {
+        for (const property of feature.properties) {
+          allProperties.push(property);
+        }
+        for (const method of feature.methods) {
+          allMethods.push(method);
+        }
+      }
+      return new Class(
+          {
+            name: 'Polymer.Base',
+            className: 'Polymer.Base',
+            properties: allProperties,
+            methods: allMethods,
+            abstract: true,
+            privacy: 'public',  // TODO(aomarks) Correct privacy?
+            // TODO(aomarks) The following should probably come from the
+            // Polymer.Base assignment instead of this final addFeature call.
```

Comment:
There's actually no documentation on (any?) of the calls we're analyzing here, so this kind of doesn't matter, and I might punt for now because it will add some complexity to track the Base assignment.

Hydrolysis actually just hard coded a description: https://github.com/Polymer/polymer-analyzer/blob/hydrolysis-1.x/src/ast-utils/docs.ts#L356. Could do the same thing here. Could also add some jsdoc to the 1.0 polymer branch.